Source/WebKit/WebProcess/InjectedBundle/API/glib/WebKitFrame.cpp

```diff
@@ -148,9 +156,6 @@ const gchar* webkit_frame_get_uri(WebKitFrame* frame)
 {
     g_return_val_if_fail(WEBKIT_IS_FRAME(frame), 0);
 
-    if (frame->priv->uri.isNull())
-        frame->priv->uri = frame->priv->webFrame->url().string().utf8();
-
```

Comment:
I think we should keep this, in case the frame is created after the loading events.

Reply:
So if we keep setting the uri here then we don't need to update it via webkitFrameSetUri, right? It will be updated for each time someone calls webkit_frame_get_uri (so we can say: "updating on demand")

Reply:
No, we keep the null check

Reply:
Ah, ok, I think I have got your point.